src/ru/multimanga/src/eu/kanade/tachiyomi/extension/ru/multimanga/MultiManga.kt

```diff
@@ -0,0 +1,442 @@
+        filters.forEach { filter ->
+            when (filter) {
+                is GenreListFilter -> {
+                    println("multi-manga: GenreListFilter: ${filter.state}")
+                    filter.state
+                        .filter { it.state != Filter.TriState.STATE_IGNORE }
+                        .forEach { genres.add(it.id) }
+                    needToBeFiltered = if (filter.state.filter { it.state != Filter.TriState.STATE_IGNORE }.isNotEmpty()) true else false
+//                    filter.state
+//                        .filter { it.state != Filter.TriState.STATE_IGNORE }
+//                        .forEach { url.addQueryParameter("genre[]", it.id) }
+                    println(
+                        "multi-manga: filter-list: ${filter.state
+                            .filter { it.state != Filter.TriState.STATE_IGNORE }}",
+                    )
+                }
+                else -> {}
+            }
+        }
+
+        val genreStr = genres.joinToString(",")
+
+        val builtUrl = if (needToBeFiltered == false) "$baseUrl/index.php?do=search&subaction=search&search_start=$page&full_search=0" else "$baseUrl/f/n.m.tags=$genreStr/sort=date/order=desc"
+        val url = builtUrl.toHttpUrl().newBuilder()
+        println("multi-manga: builtURL: $builtUrl")
+        println("multi-manga: needtoBeFiltered: $needToBeFiltered")
+        url.addQueryParameter("story", query)
+//        url.addQueryParameter("page", page.toString())
+
+        return GET(url.build(), headers)
+    }
+
+    override fun mangaDetailsParse(document: Document): SManga {
+        val manga = SManga.create()
+        manga.title = document.select("div#info-block #info h1").text()
+        manga.author = document.select("section#tags > div:contains(Автор) > span.tags").text()
+        val genres = mutableListOf<String>()
+        document.select("div:contains(Теги) > span.tags > a").forEach { element ->
+            val genre = element.text()
+            genres.add(genre)
+        }
+        manga.genre = genres.joinToString(", ")
+        manga.thumbnail_url = document.select("#cover > a > img").imgAttr()
+        return manga
+    }
+
+    private fun parseStatus(element: String): Int = when {
+        element.lowercase().contains("berjalan") -> SManga.ONGOING
+        element.lowercase().contains("tamat") -> SManga.COMPLETED
+        else -> SManga.UNKNOWN
+    }
+
+    override fun chapterListSelector() = "#content"
+
+    override fun chapterFromElement(element: Element): SChapter {
+        val urlElement = element.select("#thumbnail-container .thumb-container a").first()!!
+//        val chapter = SChapter.create()
+//        chapter.setUrlWithoutDomain(urlElement.attr("href"))
+//        chapter.name = urlElement.text()
+
+        val chapter = SChapter.create()
+        val loopbackUrl = urlElement.attr("href").removeRange(urlElement.attr("href").length - 3, urlElement.attr("href").length)
```

Comment:
`response.request.url.toString()` will give you the current url.

also for future reference, check out [StringKt documentation](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-string/)
it have methods like `substringBeforeLast`